pam/integration-tests/gdm-integration_test.go

```diff
@@ -0,0 +1,471 @@
+package main_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"slices"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/msteinert/pam/v2"
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd"
+	"github.com/ubuntu/authd/internal/testutils"
+	grouptests "github.com/ubuntu/authd/internal/users/localgroups/tests"
+	"github.com/ubuntu/authd/pam/internal/gdm"
+	"github.com/ubuntu/authd/pam/internal/gdm_test"
+	"github.com/ubuntu/authd/pam/internal/pam_test"
+	"github.com/ubuntu/authd/pam/internal/proto"
+)
+
+func init() {
+	gdm.AdvertisePamExtensions([]string{gdm.PamExtensionCustomJSON})
+}
+
+const (
+	exampleBrokerName = "ExampleBroker"
+	localBrokerName   = "local"
+	ignoredBrokerName = "<ignored-broker>"
+
+	passwordAuthID = "password"
+	fido1AuthID    = "fidodevice1"
+	phoneAck1ID    = "phoneack1"
+)
+
+func testGdmModule(t *testing.T, libPath string, args []string) {
+	t.Helper()
```

Comment:
I wouldn’t mark it as an helper here. Those are the real "tests", so if a test fails, we want to know on which line of `testGdmModule` it failed. (`t.Helper()` basically only report the caller line as the one that failed)